src/net.cpp

```diff
@@ -2337,6 +2338,16 @@ uint64_t CConnman::GetLocalServices() const
     return nLocalServices;
 }
 
+void CConnman::SetBestHeight(int height)
+{
+    nBestHeight.store(height, std::memory_order_release);
```

Comment:
I'm not sure this actually presents a bug, but it might at least exhibit some undesired behavior.

This should replace the value only if it is greater than the previous value, unless it is desirable to roll back (but this seems like a different case?).


This may also not play well with out of order headers.

In any case, I'm not sure this is a consensus critical thing, but it could be the source of future bugs or further identity leaks.

This (might) also be fixed by moving the call to SetBestHeight inside of `Lock(cs_main)` in `ActivateBestChain`. Would also be reasonable to just document (or self-doc) it a bit better about what you are expecting this to do.